go/vt/vtgate/engine/join.go

```diff
@@ -220,10 +220,10 @@ func joinFields(lfields, rfields []*querypb.Field, cols []int) []*querypb.Field
 	fields := make([]*querypb.Field, len(cols))
 	for i, index := range cols {
 		if index < 0 {
-			fields[i] = lfields[-index-1]
+			fields[i] = lfields[-index-1].CloneVT()
 			continue
 		}
-		fields[i] = rfields[index-1]
+		fields[i] = rfields[index-1].CloneVT()
```

Comment:
is there a reason to do cloning here?

Reply:
yes. if you remove this, unit tests break.

```
    join_values_test.go:100: mismatch in Fields
        [name:"col4" type:INT64 name:"col2" type:VARCHAR name:"col3" type:VARCHAR name:"col4" type:INT64 name:"col5" type:VARCHAR name:"col6" type:VARCHAR]
        want:
        [name:"col1" type:INT64 name:"col2" type:VARCHAR name:"col3" type:VARCHAR name:"col4" type:INT64 name:"col5" type:VARCHAR name:"col6" type:VARCHAR]
```

Reply:
we sometimes re-use fields, so we need to clone them. otherwise, we change the name of one field and it will impact a different field